### PR TITLE
Remove unnecessary memset() calls throughout

### PR DESCRIPTION
**Pull request by vcaputo**

### CODE REVIEW

File: sys/win32/midi-win32mm.c [hunk 1]
```diff
@@ -58,13 +58,12 @@ static unsigned int last_known_out_port = 0;
 static void _win32mm_sysex(LPMIDIHDR *q, const unsigned char *d, unsigned int len)
 {
 	char *z;
-	LPMIDIHDR m;
+	LPMIDIHDR m = {};
```
Comment (jangler) [inline sys/win32/midi-win32mm.c]: Looks like AppVeyor doesn't like this change:

```
sys/win32/midi-win32mm.c: In function '_win32mm_sysex':
sys/win32/midi-win32mm.c:61:16: error: empty scalar initializer
  LPMIDIHDR m = {};
                ^
sys/win32/midi-win32mm.c:61:16: note: (near initialization for 'm')
make[1]: *** [Makefile:1326: sys/win32/midi-win32mm.o] Error 1
make[1]: Leaving directory '/c/projects/schismtracker'
```

Comment (vcaputo) [reply sys/win32/midi-win32mm.c]: force pushed fix, thanks

File: player/csndfile.c [hunk 2]
```diff
@@ -135,7 +137,7 @@ void csf_init_instrument(song_instrument_t *ins, int samp)
 
 song_instrument_t *csf_allocate_instrument(void)
 {
-	song_instrument_t *ins = mem_calloc(1, sizeof(song_instrument_t));
+	song_instrument_t *ins = mem_alloc(sizeof(song_instrument_t));
 	csf_init_instrument(ins, 0);
```
Comment (jangler) [inline player/csndfile.c:141]: Correct me if I'm wrong, but aren't there a bunch of `song_instrument_t` fields that aren't initialized by `csf_init_instrument`?

Comment (vcaputo) [reply player/csndfile.c:141]: This commit moves the memset into `csf_init_instrument()`.
https://github.com/schismtracker/schismtracker/pull/139/files#diff-31edc9eb6dab37d1e6a5f1d7feec5b2cR123

Requiring one to zero the reused instrument prior to initialization with a memset seemed janky to me...

Comment (jangler) [reply player/csndfile.c:141]: Ah, got it. I was looking at the files in master for reference.

**Comment by jangler** [comment]
LGTM. Thanks.

**Pull request merged**